descartes_planner/src/planning_graph.cpp

```diff
@@ -51,31 +51,10 @@ PlanningGraph::~PlanningGraph()
   delete cartesian_point_link_;
 }
 
-CartesianMap PlanningGraph::getCartesianMap()
+CartesianMap PlanningGraph::getCartesianMap() const
 {
-  TrajectoryPt::ID cart_id = descartes_core::TrajectoryID::make_nil();
-  for(std::map<TrajectoryPt::ID, CartesianPointInformation>::iterator c_iter = cartesian_point_link_->begin();
-      c_iter != cartesian_point_link_->end(); c_iter++)
-  {
-    ROS_DEBUG_STREAM("Checking for last TrajectoryID: " << c_iter->first);
-    if(c_iter->second.links_.id_next.is_nil())
-    {
-      cart_id = c_iter->first;
-      break;
-    }
-  }
-
-  CartesianMap to_return = CartesianMap();
-  bool done = false;
-  while(!done)
-  {
-    to_return[cart_id] = (*cartesian_point_link_)[cart_id];
-    cart_id = (*cartesian_point_link_)[cart_id].links_.id_previous;
-    done = (cart_id.is_nil());
-    ROS_DEBUG_STREAM("Next CID: " << cart_id);
-  }
-
-  return to_return;
+  if (!cartesian_point_link_) return CartesianMap();
```

Comment:
Not sure that this needs be enclosed within __if-else__ statements as the result appears to be the same regardless of the pointer value.